app/flows/initial_request_flow.rb

```diff
@@ -10,12 +10,12 @@ def initialize(params:)
   end
 
   def template
-    if check_your_information_page?
+    if invalid_provider_search_query? || empty_search_results?
```

Comment:
Just noticing that neither `valid?` nor `validate` is being called on the form object inside of this flow. Looking at [`AllocationsController#initial_request`](https://github.com/DFE-Digital/publish-teacher-training/blob/3399-two-character-provider-search/app/controllers/providers/allocations_controller.rb#L61):

```ruby
    def initial_request
      flow = InitialRequestFlow.new(params: params)

      if request.post? && flow.valid? && flow.redirect?
        redirect_to flow.redirect_path
      else
        render flow.template, locals: flow.locals
      end
    end
```

it looks as if `template` could be called without `flow.valid?` being called, if for example `request.post?` returns `false`.

Did I read that right, and is it a bug?

Would it be worth always validating `form_object` before any of the predicates here, or in other methods, are called on it?

Reply:
Currently the validations only apply to the "Who are you requesting a course for?" page. The continue button submits a POST and validation always kicks in ... If no option is selected or 'search for a provider' is selected and the query is either blank or one character long then the validation errors will show. 

I'm not sure if this may become an issue further down the link if validations are added on other pages ... and / or whether other validations are being worked on currently. @asmega do you have a view on this?

Reply:
validation should only kick if the user submits the form. if validation occurs on a GET request of the form the user will see a load of errors on the loading of the empty form before they've filled it in